sdk/proto/agent.proto

```diff
@@ -80,6 +80,17 @@ message AgentDetails {
   repeated string tags = 3 [(gogoproto.jsontag) = "tags"];
   // Alias name for the agent
   string alias = 4 [(gogoproto.jsontag) = "alias"];
+  // Server setting for the agent
+  Server server = 5 [(gogoproto.jsontag) = "server"];
```

Comment:
I had this ready to go (just need comments to explain the settings)

```message AgentDetails {
  // List of agent feature that are enabled
  repeated string features = 1 [(gogoproto.jsontag) = "features"];
  // List of agent extensions that are enabled
  repeated string extensions = 2 [(gogoproto.jsontag) = "extensions"];
  // List of tags
  repeated string tags = 3 [(gogoproto.jsontag) = "tags"];
  // Alias name for the agent
  string alias = 4 [(gogoproto.jsontag) = "alias"];

  Server server = 5 [(gogoproto.jsontag) = "server"];
}

message Server {
    // the host name or IP of the host to connect to
	string host = 1 [(gogoproto.jsontag) = "host"];
    // the grpc port to connect to
    int32 grpc_port = 2 [(gogoproto.jsontag) = "grpc_port"];
    // a shared secrect between the server and client
    string token = 3 [(gogoproto.jsontag) = "token"];
    // metrics server name
    string metrics = 4 [(gogoproto.jsontag) = "metrics"];
    // command server name
    string command = 5 [(gogoproto.jsontag) = "command"];
    // backoff settings for exponential retry and backoff
    Backoff backoff = 6 [(gogoproto.jsontag) = "backoff"];
}

message Backoff {
    int32 intial_interval = 1 [(gogoproto.jsontag) = "intial_interval"];
    int32 randomization_factor = 2 [(gogoproto.jsontag) = "randomization_factor"];
    int32 multiplier = 3 [(gogoproto.jsontag) = "multiplier"]; 
    int32 max_interval = 4 [(gogoproto.jsontag) = "max_interval"];
    int32 max_elapsed_time = 5 [(gogoproto.jsontag) = "max_elapsed_time"];
}```

Reply:
@oliveromahony I see. Will close my MR then. 

You can tag me on your MR. Thanks

Reply:
sorry, only the changes in the proto. If you want to make them it is fine, if not I'll create the PR

Reply:
Seems like the ExpotentialBackoff library we use treats the items as int64 https://pkg.go.dev/github.com/cenkalti/backoff/v4#ExponentialBackOff - so maybe the proto needs to reflect that